bedrock/base/middleware.py

```diff
@@ -97,3 +98,20 @@ def process_request(self, request):
 
 class FrameOptionsHeader(OldFrameOptionsHeader, MiddlewareMixin):
     pass
+
+
+class MetricsStatusMiddleware(MiddlewareMixin):
+    """Send status code counts to statsd"""
+
+    def _record(self, status_code):
+        metrics.incr("response.status", tags=[f"status_code:{status_code}"])
+
+    def process_response(self, request, response):
+        self._record(response.status_code)
+        return response
+
+    def process_exception(self, request, exception):
+        if isinstance(exception, Http404):
+            self._record(404)
+        else:
+            self._record(500)
```

Comment:
I wondered whether this is too much of a catch-all, but then I remembered we don't need to issue 401s and 403s in Bedrock

Reply:
That's true though. If a more correct implementation would be to check for other exceptions here and pass along the appropriate status, I should add it here. I'll check it out.

Reply:
I looked deeper into this. All of the other `Http*` response classes are returned, not raised. So that would get processed in `process_response`. So the `Http404` is the only one we need to catch here.

Reply:
I did find some exceptions elsewhere in the Django code however, so I can add the ones handled in [response_for_exception](https://github.com/django/django/blob/main/django/core/handlers/exception.py#L63) here as well.

Reply:
Because some exceptions are handled in `response_for_exception` handler, they come through as responses instead of exceptions in the middleware. Interesting. So I added tests for that but the middleware code itself has no change.